### ISSUE TITLE
problem compiling fiberassign at KPNO

### ISSUE THREAD

**Issue by sbailey** [desihub/fiberassign#272]
I'm having difficulty compiling fiberassign at KPNO (full compile error below).

I suspect this is related to a rather old default gcc:
```
datasystems@desi-7 [fiberassign-1.4.1]$ which gcc
/usr/bin/gcc
datasystems@desi-7 [fiberassign-1.4.1]$ gcc --version
gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-36)
```
vs. gcc 8.3.0 at NERSC.

Is there as simple workaround, or is the best solution to request a gcc upgrade?  gcc upgrades could have lots of other effects for the online systems, so I wouldn't make that request lightly.

@tskisner help?  thanks...

Compile messages:
```
datasystems@desi-7 [fiberassign-1.4.1]$ python setup.py clean; python setup.py build_ext --inplace
running clean
removing 'build/temp.linux-x86_64-3.8' (and everything under it)
running build_ext
gcc -pthread -B /software/datasystems/desiconda/20200924/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/software/datasystems/desiconda/20200924/conda/include/python3.8 -c /tmp/tmpcv_hm4_r.cpp -o tmp/tmpcv_hm4_r.o -std=c++14
gcc -pthread -B /software/datasystems/desiconda/20200924/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/software/datasystems/desiconda/20200924/conda/include/python3.8 -c /tmp/tmpiecknb9_.cpp -o tmp/tmpiecknb9_.o -std=c++11
gcc -pthread -B /software/datasystems/desiconda/20200924/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/software/datasystems/desiconda/20200924/conda/include/python3.8 -c /tmp/tmpjhlkbwle.cpp -o tmp/tmpjhlkbwle.o -fvisibility=hidden
gcc -pthread -B /software/datasystems/desiconda/20200924/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -I/software/datasystems/desiconda/20200924/conda/include/python3.8 -c /tmp/tmpfwrzc71i.cpp -o tmp/tmpfwrzc71i.o -fopenmp
building 'fiberassign._internal' extension
creating build
creating build/temp.linux-x86_64-3.8
creating build/temp.linux-x86_64-3.8/src
gcc -pthread -B /software/datasystems/desiconda/20200924/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Isrc -I/software/datasystems/desiconda/20200924/conda/include/python3.8 -c src/utils.cpp -o build/temp.linux-x86_64-3.8/src/utils.o -DVERSION_INFO="1.4.1" -std=c++11 -fvisibility=hidden -fopenmp
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
gcc -pthread -B /software/datasystems/desiconda/20200924/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Isrc -I/software/datasystems/desiconda/20200924/conda/include/python3.8 -c src/hardware.cpp -o build/temp.linux-x86_64-3.8/src/hardware.o -DVERSION_INFO="1.4.1" -std=c++11 -fvisibility=hidden -fopenmp
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
gcc -pthread -B /software/datasystems/desiconda/20200924/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Isrc -I/software/datasystems/desiconda/20200924/conda/include/python3.8 -c src/tiles.cpp -o build/temp.linux-x86_64-3.8/src/tiles.o -DVERSION_INFO="1.4.1" -std=c++11 -fvisibility=hidden -fopenmp
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
gcc -pthread -B /software/datasystems/desiconda/20200924/conda/compiler_compat -Wl,--sysroot=/ -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -Isrc -I/software/datasystems/desiconda/20200924/conda/include/python3.8 -c src/targets.cpp -o build/temp.linux-x86_64-3.8/src/targets.o -DVERSION_INFO="1.4.1" -std=c++11 -fvisibility=hidden -fopenmp
cc1plus: warning: command line option ‘-Wstrict-prototypes’ is valid for C/ObjC but not for C++ [enabled by default]
src/targets.cpp: In constructor ‘fiberassign::TargetTree::TargetTree(fiberassign::Targets::pshr, double)’:
src/targets.cpp:232:13: error: ‘make_unique’ is not a member of ‘std’
     tree_ = std::make_unique < htmTree <TreePoint> > (treelist_, mintreesz_);
             ^
src/targets.cpp:232:52: error: expected primary-expression before ‘>’ token
     tree_ = std::make_unique < htmTree <TreePoint> > (treelist_, mintreesz_);
                                                    ^
src/targets.cpp:232:66: warning: left operand of comma operator has no effect [-Wunused-value]
     tree_ = std::make_unique < htmTree <TreePoint> > (treelist_, mintreesz_);
                                                                  ^
error: command 'gcc' failed with exit status 1
```

**Comment by tskisner** [comment]
Redhat ships an ancient compiler so that binaries built with that can run across multiple redhat versions.  However, most people I know install a newer compiler from redhat packages.

Having said that, this is actually an error on my end.  Apparently `std::make_unique` was not formally added to the standard until C++14.  Since the intention was to only require C++11, I should fix that.  Will try to get a PR open later this morning.

**Comment by tskisner** [comment]
Ok, I tested the fix in #273 inside a docker container running CentOS-7 (which I guessed is the same as KPNO, from the gcc version 4.8.5).  I reproduce your error without that fix and then fiberassign builds fine with the fix.
